### DISCUSSION TITLE
Potentially increase `maxDepth` of Gradle (Groovy) parser

### DISCUSSION THREAD

**Discussion by mattmapadmi** [renovatebot/renovate#32513] [Suggest an Idea]
### Tell us more.

I've inherited an Android project with an obnoxiously large app-level build.gradle (400+ lines, with a 188 line Flavours block), and when adding Renovate, I've found that it is unable to find deprecated packages within the `dependencies` block at the bottom of the file. It will find things in the `plugins` block at the top, and if I move my `dependencies` block above the `android` block, that also works.

Almost by pure fluke, I found that by changing `maxDepth` to 48 in the parser config, it starts working again.

https://github.com/renovatebot/renovate/blob/40a600488c455aeec7132ce73635daa01c4d098c/lib/modules/manager/gradle/parser.ts#L54

Looking at the good-enough-parser's defaults, they set it to 1024 by default, so 32 seems to be very low.

https://github.com/zharinov/good-enough-parser/blob/2bd0a2f2d3b04d2c3572d8efebe362cdcc8a815b/lib/query/matchers/tree-matcher.ts#L39

I know the solution on my side is to tidy up the config (and use a Version Catalog), but I just wanted to flag it, as it fails in an incredibly vague way, and it may be something not taken into account when setting the `maxDepth` originally.

**Comment by rarkins** [comment]
@zharinov
